src/include/duckdb/main/connection.hpp

```diff
@@ -55,6 +56,9 @@ class Connection {
 	//! Returns query profiling information for the current query
 	DUCKDB_API string GetProfilingInformation(ProfilerPrintFormat format = ProfilerPrintFormat::QUERY_TREE);
 
+	//! Returns the first node of the query profiling tree
+	DUCKDB_API ProfilingNode *GetProfilingTree();
```

Comment:
As this is still within C++-land, we likely want to return `optional_ptr` or (better, if possible) a reference here.